error: use of overloaded operator '<<' is ambiguous

```
#pythran export test1 ((float, float) list)
#pythran export test1 ((int, int) list)
def test1 (arg1):
    print ([(type(x[0]), type(x[1])) for x in arg1])
```

This gives:
```
/home/nbecker/.local/lib/python3.7/site-packages/pythran/pythonic/types/tuple.hpp:383:12: error: use of overloaded operator '<<' is ambiguous (with operand types 'std::ostream' (aka 'basic_ostream<char>') and 'const (anonymous namespace)::pythonic::__builtin__::functor::float_')
        os << *iter++ << ", ";
```

gcc-9.2.1-1.fc31.x86_64
same with clang-9.0.0-1.fc31.x86_64

--- a/pythran/pythonic/include/__builtin__/complex.hpp
+++ b/pythran/pythonic/include/__builtin__/complex.hpp
@@ -14,8 +14,12 @@ namespace __builtin__
     struct complex {
       using callable = void;
       using type = std::complex<double>;
-      // TODO: do ! handle string as first argument
+      // TODO: doesn't handle string as first argument
       type operator()(double v0 = 0, double v1 = 0) const;
+      friend std::ostream &operator<<(std::ostream &os, complex)
+      {
+        return os << "complex";
+      }
     };
   }
 }

--- a/pythran/pythonic/include/__builtin__/float_.hpp
+++ b/pythran/pythonic/include/__builtin__/float_.hpp
@@ -18,6 +18,11 @@ namespace __builtin__
       type operator()(T &&t) const;
 
       type operator()() const;
+
+      friend std::ostream &operator<<(std::ostream &os, float_)
+      {
+        return os << "float";
+      }
     };
   }
 }

--- a/pythran/tests/notebooks/export.ipynb
+++ b/pythran/tests/notebooks/export.ipynb
@@ -878,6 +878,46 @@
     "y = np.ones(5)[None, :]\n",
     "views(x), views(y)"
    ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "When usure of an object type, just print it!"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 43,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%pythran\n",
+    "#pythran export dump_type((float, bool) list)\n",
+    "#pythran export dump_type((int, complex) list)\n",
+    "def dump_type(arg1):\n",
+    "    return str([(type(x[0]), type(x[1])) for x in arg1])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 44,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "('[(float, bool)]', '[(int64, complex)]')"
+      ]
+     },
+     "execution_count": 44,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "dump_type([(1., True)]), dump_type([(1, 1j)])"
+   ]
   }
  ],
  "metadata": {

